### DISCUSSION TITLE
Keep stack leniency consistent throughout all mod combinations on the same map

### DISCUSSION THREAD

**Discussion by Kingling664** [ppy/osu#13144] [Ideas] [answered]
Currently with how osu! stable works, putting on mods like HR can change how circles are stacked. This means circles that the mapper intended to be stacked normally can become perfect stacks which impacts the playability of the map. Perfect stacks only really work when the mapper explicitly intended them to be perfect stacks, as they can often cause some maps to be unreadable and notes stacked seemingly randomly when not intended. 

My proposal is for all note stacking to be the same as how notes are stacked on nomod, regardless of mods added.

**Comment by Karthyyyy** [comment]
Only just found out this existed, got pointed this way. Doesn't make any sense for stack leniency to be tied to AR from a gameplay perspective. Perfect stacks are often a gimmick incorporated by mappers, it's frustrating to be playing a normal map, add HR and all the stacks become perfectly stacked just because you put HR on. Similarly I imagine it's equally as frustrating for mappers trying to make their stacks not become perfectly stacked when HR is added.

**Comment by peppy** [comment]
100% agree with this. the fact it changes is a bug which we should take this opportunity to fix.

**Reply by bdach** [reply, marked as answer]
If it is a bug, then let's track it as a bug at ~~https://github.com/ppy/osu/issues/23652~~ https://github.com/ppy/osu/issues/14811.

**Comment by bdach** [comment]
What my biggest worry is - should the old/broken behaviour still be accessible by putting classic mod on? That implies that HR + classic would need to have a unique effect when combined...

**Reply by peppy** [reply]
Honestly for this one I think it should be fixed in all cases at the cost of breaking some replays. I don't think it change difficulty enough to matter, but feel free to bring up specific cases if they exist.

**Comment by Wieku** [comment]
So from my understanding it should work based on map's original AR? I was thinking if stack threshold and spread ratio (or even direction) as a timing point parameter would be of any use to mappers (like similar object group should be perfectly stacked at one section but spaced differently in another and at different angle)

**Reply by peppy** [reply]
let's not make this more complex than it needs to be.
